### PR TITLE
feat(core): save user habits in right sidebar

### LINKED ISSUE #6237: Remember state of right sidebar
### Description

Currently the state of the right side bar is not persisted between app restarts.
I'd like the table of contents to always be visible, which means every time I open the app I have to
1. Click the button to open the right sidebar
2. Switch from the default calendar view to the table of contents

### Use case

Anyone who uses any of the three sidebar panels a lot would profit from the sidebar remembering whether it was open and which panel was last selected.

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by fundon**
Closes #6237

### CODE REVIEW

**Comment by fundon** [comment]
* **#6262** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6262?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fundon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/modules/right-sidebar/entities/right-sidebar.ts [hunk 1]
```diff
@@ -1,9 +1,13 @@
 import { LiveData } from '@toeverything/infra/livedata';
 
+import { LocalStorageGlobalState } from '../../infra-web/storage';
 import type { RightSidebarView } from './right-sidebar-view';
 
 export class RightSidebar {
-  readonly isOpen = new LiveData(false);
+  private readonly globalState = new LocalStorageGlobalState();
```
Comment (EYHN) [inline packages/frontend/core/src/modules/right-sidebar/entities/right-sidebar.ts]:
```suggestion
  constructor(private readonly globalState: GlobalState) {}
```

File: packages/frontend/core/src/modules/right-sidebar/entities/right-sidebar.ts [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
+  readonly isOpen = new LiveData(
+    Boolean(this.globalState.get<boolean>('app:settings:rightsidebar'))
+  );
```
Comment (EYHN) [inline packages/frontend/core/src/modules/right-sidebar/entities/right-sidebar.ts]:
```suggestion
  readonly isOpen = this.globalState.watch<boolean>('app:settings:rightsidebar').map(f => !!(f ?? false));
```

File: packages/frontend/core/src/modules/right-sidebar/entities/right-sidebar.ts [hunk 3]
```diff
@@ -12,14 +16,18 @@ export class RightSidebar {
...
   open() {
     this.isOpen.next(true);
+    this.globalState.set('app:settings:rightsidebar', true);
   }
 
   toggle() {
-    this.isOpen.next(!this.isOpen.value);
+    const value = !this.isOpen.value;
+    this.isOpen.next(value);
+    this.globalState.set('app:settings:rightsidebar', value);
```
Comment (EYHN) [inline packages/frontend/core/src/modules/right-sidebar/entities/right-sidebar.ts]:
```suggestion
    const value = !this.isOpen.value;
    this.globalState.set('app:settings:rightsidebar', value);
```

File: packages/frontend/core/src/modules/right-sidebar/entities/right-sidebar.ts [hunk 4]
```diff
@@ -12,14 +16,18 @@ export class RightSidebar {
...
   }
 
   close() {
     this.isOpen.next(false);
+    this.globalState.set('app:settings:rightsidebar', false);
```
Comment (EYHN) [inline packages/frontend/core/src/modules/right-sidebar/entities/right-sidebar.ts]:
```suggestion
    this.globalState.set('app:settings:rightsidebar', false);
```

File: packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx [hunk 5]
```diff
@@ -66,6 +67,10 @@ import * as styles from './detail-page.css';
 import { DetailPageHeader } from './detail-page-header';
 
 const DetailPageImpl = memo(function DetailPageImpl() {
+  const globalState = useService(GlobalState);
+  const activeTab = globalState.get<SidebarTabName>(
+    'app:settings:rightsidebar:tabs:active'
+  );
```
Comment (EYHN) [inline packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx]:
```suggestion
  const activeTab = useLiveData(globalState.watch<SidebarTabName>(
    'app:settings:rightsidebar:tabs:active'
  ));
```

File: packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx [hunk 6]
```diff
@@ -85,9 +90,13 @@ const DetailPageImpl = memo(function DetailPageImpl() {
   }, [editor, isActiveView, setActiveBlockSuiteEditor]);
 
   const [activeTabName, setActiveTabName] = useState<SidebarTabName | null>(
```
Comment (EYHN) [inline packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx]: we can delete this state, just use globalstate

**Review by EYHN — CHANGES_REQUESTED** [review]

File: packages/frontend/core/src/modules/right-sidebar/entities/right-sidebar.ts (hunk 2 above)
Comment (fundon) [reply packages/frontend/core/src/modules/right-sidebar/entities/right-sidebar.ts]: `this.globalState.watch(...).map is not a function`

**Review by EYHN — APPROVED** [review]

**Pull request merged**